packages/opentelemetry-tracing/src/config.ts

```diff
@@ -23,10 +33,78 @@ import { AlwaysOnSampler, getEnv } from '@opentelemetry/core';
  * used to extend the default value.
  */
 export const DEFAULT_CONFIG = {
-  sampler: new AlwaysOnSampler(),
+  sampler: buildSamplerFromEnv(env),
   traceParams: {
     numberOfAttributesPerSpan: getEnv().OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT,
     numberOfLinksPerSpan: getEnv().OTEL_SPAN_LINK_COUNT_LIMIT,
     numberOfEventsPerSpan: getEnv().OTEL_SPAN_EVENT_COUNT_LIMIT,
   },
 };
+
+/**
+ * Based on environment, builds a sampler, complies with specification.
+ * @param env optional, by default uses getEnv(), but allows passing a value to reuse parsed environment
+ */
+export function buildSamplerFromEnv(
+  env: Required<ENVIRONMENT> = getEnv()
+): Sampler {
+  switch (env.OTEL_TRACES_SAMPLER) {
```

Comment:
all these values `always_on` etc. should be some enum / const defined in one place and then just please reference this everywhere (including logs, descriptions etc.) to avoid error, typos etc..
```
enum OTEL_TRACES_SAMPLER_VALUES = {
  ALWAYS_ON: 'always_on',
  ALWAYS_OFF: 'always_off',
....
}
```
and then
```
case OTEL_TRACES_SAMPLER_VALUES.ALWAYS_ON: 
...
```